src/trackinfoobject.h

```diff
@@ -32,10 +32,27 @@ class TrackInfoObject : public QObject {
     Q_OBJECT
 
   public:
-    // Initialize track with a QFileInfo class
-    explicit TrackInfoObject(
+    TrackInfoObject(const TrackInfoObject&) = delete;
+    TrackInfoObject(TrackInfoObject&&) = delete;
```

Comment:
I will replace all pass-by-value/move parameters with const-ref/copy until 
we have common rules whether to copy or move arguments.

There are already some recommendations in the C++ Core Guidelines on how to 
pass parameters:

F.15: Prefer simple and conventional ways of passing information 
<https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rf-conventional>

https://github.com/isocpp/CppCoreGuidelines

Some of the rules (F.17) differ from the current Mixxx rules.

On 02/20/2016 10:40 PM, Daniel Schürmann wrote:
>
> In src/trackinfoobject.h 
> <https://github.com/mixxxdj/mixxx/pull/898#discussion_r53557615>:
>
> > @@ -32,10 +32,27 @@ class TrackInfoObject : public QObject {
> >      Q_OBJECT
> >
> >    public:
> > -    // Initialize track with a QFileInfo class
> > -    explicit TrackInfoObject(
> > +    TrackInfoObject(const TrackInfoObject&) = delete;
> > +    TrackInfoObject(TrackInfoObject&&) = delete;
>
> This is redundant since the compiler does not create a move constructors 
> if the copy constructor is deleted.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/898/files#r53557615>.
>